gadgets/trace_tcp/ebpf/close.h

```diff
@@ -26,19 +43,35 @@ static __always_inline void handle_tcp_close(struct pt_regs *ctx,
 	u8 oldstate = BPF_CORE_READ(sk, __sk_common.skc_state);
 	if (oldstate == TCP_SYN_SENT || oldstate == TCP_SYN_RECV ||
 	    oldstate == TCP_NEW_SYN_RECV)
-		return;
+		goto end;
 
 	family = BPF_CORE_READ(sk, __sk_common.skc_family);
 	if (!fill_tuple(&tuple, sk, family))
-		return;
+		goto end;
 
 	event = gadget_reserve_buf(&events, sizeof(*event));
 	if (!event)
-		return;
+		goto end;
 
 	fill_event(event, &tuple, NULL, 0, close);
+	event->new_fd = 0;
+
+	fd = bpf_map_lookup_elem(&tcp_tid_fd, &tid);
+	// Best effor approach to see if we can find it through the sock
+	if (!fd)
+		fd = bpf_map_lookup_elem(&tcp_sock_fd, &sk);
+
+	if (fd)
+		event->fd = *fd;
+	else {
+		// We only have a fd when we go through sys_close
+		// and we should not discard the event in the other case
+		event->fd = 0;
```

Comment:
Use -1 to distinguish "unknown fd" and "fd 0". Change `if (fd)` accordingly.
And change u32 type to (signed) int.